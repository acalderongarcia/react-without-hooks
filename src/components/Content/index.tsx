import React from 'react';
import PropTypes from "prop-types";

//interface Meh{children?: ReactNode}
//interface Props{body: Meh}

class Content extends React.Component{
    static propTypes = {
        body: PropTypes.object.isRequired
    };

    render() {
        const body = this.props;
        return (
            <div className="Content">
                {body.children}
            </div>
        );
    }
}

export default Content;
