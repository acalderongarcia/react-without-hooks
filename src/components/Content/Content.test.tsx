import React from 'reactn';
import renderer from 'react-test-renderer';
import 'bootstrap/dist/css/bootstrap.css';
import Content from './index';

describe('Testing src/components/Content/index', () => {

    test('Content it renders', () => {
        // Before
        const wrapper = renderer.create(
            <div className="Content">
                {Content.propTypes.body.children}
            </div>
        );
        // Then
        let tree = wrapper.toJSON();
        // After
        expect(tree).toMatchSnapshot();
    });
});




