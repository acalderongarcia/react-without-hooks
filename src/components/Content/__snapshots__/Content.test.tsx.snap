// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing src/components/Content/index Content it renders 1`] = `
<div
  className="Content"
/>
`;
